### PR TITLE
Get real BT MAC Address from /sys/kernel

### PR DESCRIPTION
**Pull request by jp-bennett**
Portduino isn't returning a valid MAC Address, even though systems like the Raspberry Pi do have Bluetooth MAC we could use.

### CODE REVIEW

**Comment by jp-bennett** [comment]
I know this needs some added error-checking, but could somebody that knows portduino better give it a sanity check? Does this make sense?

**Review by caveman99 — CHANGES_REQUESTED** [review]
Portduino is used on linux in general, not only raspberry type computers. can you provide both means of obtaining the mac address? E.G. only fall back to the raspberry way when the portduino config does not contain a Mac?

**Comment by jp-bennett** [comment]
> Portduino is used on linux in general, not only raspberry type computers. can you provide both means of obtaining the mac address? E.G. only fall back to the raspberry way when the portduino config does not contain a Mac?

OK, so the logic should be that we use the MAC from the config if possible, and if empty we try to pull a Bluetooth MAC? Should be doable. And I think it could work on any machine with a Bluetooth adapter.

**Comment by caveman99** [comment]
Yes, if there's a mac in the config, it has priority.

File: cores/portduino/main.cpp [hunk 1]
```diff
@@ -75,12 +78,18 @@ static error_t parse_opt(int key, char *arg, struct argp_state *state) {
...
-  dmac[2] = portduinoArguments.hwId >> 24;
-  dmac[3] = portduinoArguments.hwId >> 16;
-  dmac[4] = portduinoArguments.hwId >> 8;
-  dmac[5] = portduinoArguments.hwId & 0xff;
+    std::fstream macIdentity;
+    macIdentity.open("/sys/kernel/debug/bluetooth/hci0/identity", std::ios::in);
+    std::string macLine;
+    getline(macIdentity, macLine);
+    macIdentity.close();
+
```
Comment (caveman99) [inline cores/portduino/main.cpp:86]: better check if that sys node actually exists. There are linux systems without bluetooth support...

**Review by caveman99 — CHANGES_REQUESTED** [review]

File: cores/portduino/main.cpp [hunk 2]
```diff
@@ -75,12 +78,18 @@ static error_t parse_opt(int key, char *arg, struct argp_state *state) {
...
 
 void getMacAddr(uint8_t *dmac) {
-  dmac[0] = 0x80;
-  dmac[1] = 0;
-  dmac[2] = portduinoArguments.hwId >> 24;
-  dmac[3] = portduinoArguments.hwId >> 16;
-  dmac[4] = portduinoArguments.hwId >> 8;
-  dmac[5] = portduinoArguments.hwId & 0xff;
+    std::fstream macIdentity;
+    macIdentity.open("/sys/kernel/debug/bluetooth/hci0/identity", std::ios::in);
```
Comment (mverch67) [inline cores/portduino/main.cpp:82]: There should be a way do retrieve the bluetooth address without root rights (e.g. see hciconfig). meshtaticd should not require sudo permissions for running.

Comment (jp-bennett) [reply cores/portduino/main.cpp]: This now runs without root, and fails gracefully.

**Review by thebentern — APPROVED** [review]

**Pull request merged**